internal/db/update/schema.go

```diff
@@ -126,25 +128,30 @@ func (s *SchemaUpdate) Ensure(db *sql.DB) (int, error) {
 func (s *SchemaUpdate) Dump(db *sql.DB) (string, error) {
 	var statements []string
 	err := query.Transaction(context.TODO(), db, func(ctx context.Context, tx *sql.Tx) error {
-		versions, err := query.SelectIntegers(ctx, tx, "SELECT version FROM schemas ORDER BY version")
-		if err != nil {
-			return err
-		}
+		current := 0
+		for _, versionType := range []int{0, 1} {
+			versions, err := query.SelectIntegers(ctx, tx, "SELECT version FROM schemas WHERE type = ? ORDER BY version", versionType)
+			if err != nil {
+				return err
+			}
 
-		if len(versions) == 0 {
-			return fmt.Errorf("expected schema table to contain at least one row")
-		}
+			if len(versions) == 0 {
+				return fmt.Errorf("expected schema table to contain at least one row")
+			}
 
-		err = checkSchemaVersionsHaveNoHoles(versions)
-		if err != nil {
-			return err
+			err = checkSchemaVersionsHaveNoHoles(versions)
+			if err != nil {
+				return err
+			}
+
+			current += versions[len(versions)-1]
 		}
 
-		current := versions[len(versions)-1]
-		if current != len(s.updates) {
-			return fmt.Errorf("Update level is %d, expected %d", current, len(s.updates))
+		if current != len(s.internalUpdates)+len(s.externalUpdates) {
```

Comment:
Would it be more helpful to do this check inside the for loop for each schema version type so that the error can indicate which schema type is having problems?